src/azure-cli/azure/cli/command_modules/appconfig/_utils.py

```diff
@@ -34,19 +34,51 @@ def construct_connection_string(cmd, config_store_name):
 
 
 def resolve_store_metadata(cmd, config_store_name):
+    resource_group = None
+    endpoint = None
     try:
         config_store_client = cf_configstore(cmd.cli_ctx)
         all_stores = config_store_client.list()
         for store in all_stores:
             if store.name.lower() == config_store_name.lower():
-                # Id has a fixed structure /subscriptions/subscriptionName/resourceGroups/groupName/providers/providerName/configurationStores/storeName"
-                return store.id.split('/')[4], store.endpoint
+                if resource_group is not None:
+                    raise CLIError('Multiple configuration stores found with name {}.'.format(config_store_name))
+                else:
+                    # Id has a fixed structure /subscriptions/subscriptionName/resourceGroups/groupName/providers/providerName/configurationStores/storeName"
+                    resource_group = store.id.split('/')[4]
+                    endpoint = store.endpoint
     except HttpResponseError as ex:
         raise CLIError("Failed to get the list of App Configuration stores for the current user. Make sure that the account that logged in has sufficient permissions to access the App Configuration store.\n{}".format(str(ex)))
 
+    if resource_group is not None and endpoint is not None:
+        return resource_group, endpoint
+
     raise CLIError("Failed to find the App Configuration store '{}'.".format(config_store_name))
 
 
+def resolve_deleted_store_metadata(cmd, config_store_name):
+    resource_group = None
+    location = None
+    try:
+        client = cf_configstore(cmd.cli_ctx)
+        deleted_stores = client.list_deleted()
+        for deleted_store in deleted_stores:
+            if deleted_store.name.lower() == config_store_name.lower():
+                if location is not None:
+                    raise CLIError('Multiple configuration stores found with name {}.'.format(config_store_name))
+                else:
```

Comment:
Unnecessary "else" after "raise" (no-else-raise)
```suggestion
                if location is None:
```